### ISSUE TITLE
Monitor serial numbers disappeared

### ISSUE THREAD

**Issue by gnull** [swaywm/sway#6602]
- **Sway Version:** `sway version 1.6-f4db502d (Oct 10 2021, branch 'master')`
- **Debug Log:** [gist](https://gist.github.com/gnull/bce820414013bf34e3b6d75a27451b2d)
- **Configuration File:** default

I have two monitors of the same model connected to my laptop. Before, the monitors used to have unique names and I could reliably configure them in Sway and Kanshi. But recently I noticed that the last parts of their names (the serial numbers — I just checked, they match the values printed at the back of the monitors) disappeared.

Old names were `Goldstar Company Ltd LG IPS FULLHD 604NTDV33634` and `Goldstar Company Ltd LG IPS FULLHD 604NTCZ33627`, now both became `Goldstar Company Ltd LG IPS FULLHD `. Now my `swaymsg -t get_outputs` outputs this:

https://gist.github.com/gnull/ce1d91e08b6e9dc820df5da214fae44e

Is this a Sway issue? Any ideas how I could bring back my unique serial numbers or further debug this? Where does Sway normally obtain the serial numbers from?

Around the time I started having the issue I had reset both monitors to factory settings, and I also updated all packages on my Arch System. I wonder if any of these could cause the issue.

**Comment by emersion** [comment]
The serial number if fetched from the monitor's EDID. Maybe try running `edid-decode` and check if the serial is in there. We haven't changed the logic for this in a long time.

**Comment by gnull** [comment]
For anyone else struggling with a similar issue: my problem was caused by the new cables I got. Apparently, the cables I connect my monitors with are messing with monitors' EDID info and among other things stripping the serial number info. One cable is a simple HDMI, the other one is a DP-to-HDMI.

I replaced one of them with a DVI cable, and now both Sway and `edid-decode` see the corresponding monitor's serial number. For me, it's a workaround since now my setup can tell the monitors from each other.
